mobile_config/src/gateway/db.rs

```diff
@@ -0,0 +1,365 @@
+            r#"
+            SELECT
+                address,
+                gateway_type,
+                created_at,
+                updated_at,
+                refreshed_at,
+                last_changed_at,
+                hash,
+                antenna,
+                elevation,
+                azimuth,
+                location,
+                location_changed_at,
+                location_asserts
+            FROM gateways
+            WHERE address = $1
+            "#,
+        )
+        .bind(address.as_ref())
+        .fetch_optional(db)
+        .await?;
+
+        Ok(gateway)
+    }
+
+    pub fn stream_by_addresses<'a>(
+        db: impl PgExecutor<'a> + 'a,
+        addresses: &[PublicKeyBinary],
+        min_updated_at: DateTime<Utc>,
+    ) -> impl Stream<Item = Self> + 'a {
+        let addr_array: Vec<Vec<u8>> = addresses.iter().map(|a| a.as_ref().to_vec()).collect();
+
+        sqlx::query_as::<_, Self>(
+            r#"
+            SELECT
+                address,
+                gateway_type,
+                created_at,
+                updated_at,
+                refreshed_at,
+                last_changed_at,
+                hash,
+                antenna,
+                elevation,
+                azimuth,
+                location,
+                location_changed_at,
+                location_asserts
+            FROM gateways
+            WHERE address = ANY($1)
+                AND (updated_at >= $2 OR refreshed_at >= $2 OR created_at >= $2)
+            "#,
+        )
+        .bind(addr_array)
+        .bind(min_updated_at)
+        .fetch(db)
+        .map_err(anyhow::Error::from)
+        .filter_map(|res| async move { res.ok() })
+    }
+
+    pub fn stream_by_types<'a>(
+        db: impl PgExecutor<'a> + 'a,
+        types: Vec<GatewayType>,
+        min_date: DateTime<Utc>,
+        min_location_changed_at: Option<DateTime<Utc>>,
+    ) -> impl Stream<Item = Self> + 'a {
+        sqlx::query_as::<_, Self>(
+            r#"
+                SELECT
+                    address,
+                    gateway_type,
+                    created_at,
+                    updated_at,
+                    refreshed_at,
+                    last_changed_at,
+                    hash,
+                    antenna,
+                    elevation,
+                    azimuth,
+                    location,
+                    location_changed_at,
+                    location_asserts
+                FROM gateways
+                WHERE gateway_type = ANY($1)
+                AND (updated_at >= $2 OR refreshed_at >= $2 OR created_at >= $2)
+                AND (
+                    $3::timestamptz IS NULL
+                    OR (location IS NOT NULL AND location_changed_at >= $3)
+                )
+            "#,
+        )
+        .bind(types)
+        .bind(min_date)
+        .bind(min_location_changed_at)
+        .fetch(db)
+        .map_err(anyhow::Error::from)
```

Comment:
Should we be logging here or adding some metrics to know if our db is starting to fill with rows we can't decode?

Reply:
IMO these should not happen or they would explode during insertion so I don't think so